### ISSUE TITLE
`on()` should work

### ISSUE THREAD

**Issue by brian-brazil** [prometheus/prometheus#1597]
We should accept an empty list of labels for aggregation and vector matching. The only place this offers new semantics is with `on`, as it'll let us do `on()` rather than `on(dummy)`.

**Comment by beorn7** [comment]
Would `on` be valid, too, then? I.e. `on` == `on()` == `on(does_not_exist)` ?

**Comment by brian-brazil** [comment]
Hmm, for clarity we may want to keep it to only `on()`

**Comment by juliusv** [comment]
:+1: to always requiring the parentheses.

**Comment by brian-brazil** [comment]
Should we also require them for `group_left` for consistency? My instinct says no.

**Comment by beorn7** [comment]
From a natural language perspective, `group_left`/`group_right` totally makes sense without parentheses, while `on` would imply that the next token is what you join "on". From a purely syntactical perspective, it's however inconsistent that sometimes `modifier()` is equivalent to `modifier` and sometimes not.

Another point is that `group_left`/`group_right` will be used most of the time without any following labels, while it's the other way round with `on` (usually you would list at least one label, and the case without a label is pretty special).

In conclusion, I'd definitely allow `group_left`/`group_right` without parentheses. Whether we want allow `on` without parentheses for consistency or require `on()` to avoid misreading of queries is a matter of discussion.

**Comment by fabxc** [comment]
What's the status here?
How necessary is it really for 1.0?

**Comment by brian-brazil** [comment]
I hope to take care of this in the next few weeks.

This isn't critical for 1.0, it cleans up a not uncommon use case.
